perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_loa.java

```diff
@@ -78,18 +78,24 @@ public AttributeDefinition getAttributeDefinition() {
 		List<AuditEvent> resolvingMessages = new ArrayList<>();
 		if (message == null) return resolvingMessages;
 
-		if (message instanceof UserExtSourceAddedToUser) {
-			resolvingMessages.add(resolveEvent(sess, ((UserExtSourceAddedToUser) message).getUser()));
-		} else if (message instanceof UserExtSourceRemovedFromUser) {
-			resolvingMessages.add(resolveEvent(sess, ((UserExtSourceRemovedFromUser) message).getUser()));
-		} else if (message instanceof UserExtSourceUpdated) {
-			try {
+		User user = null;
+		try {
+			if (message instanceof UserExtSourceAddedToUser) {
+				user = ((UserExtSourceAddedToUser) message).getUser();
+				sess.getPerunBl().getUsersManagerBl().checkUserExists(sess, user);
+				resolvingMessages.add(resolveEvent(sess, user));
+			} else if (message instanceof UserExtSourceRemovedFromUser) {
+				user = ((UserExtSourceRemovedFromUser) message).getUser();
+				sess.getPerunBl().getUsersManagerBl().checkUserExists(sess, user);
+				resolvingMessages.add(resolveEvent(sess, user));
+			} else if (message instanceof UserExtSourceUpdated) {
 				resolvingMessages.add(resolveEvent(sess, sess.getPerunBl().getUsersManagerBl().getUserById(
-						sess, ((UserExtSourceUpdated) message).getUserExtSource().getUserId())));
-			} catch (UserNotExistsException e) {
-				throw new ConsistencyErrorException("User associated with updated UserExtSource no longer exists while resolving virtual attribute value change.", e);
+							sess, ((UserExtSourceUpdated) message).getUserExtSource().getUserId())));
 			}
+		} catch (UserNotExistsException e) {
+			throw new ConsistencyErrorException("User "+user+" associated with "+message.getName()+" event no longer exists while resolving virtual attribute value change for LoA.", e);
```

Comment:
We can ignore this situation, just log it instead of throwing an exception.

Reply:
Fixed, now message is logged and no exception is thrown.